pallets/pallet-bonded-coins/README.md

```diff
@@ -1,4 +1,137 @@
+
+More information and implementation details can be found [here][bonding-curve].
+
+### Collateral
+Collateral is an existing currency that must be locked up to mint new tokens. The amount of collateral required is determined by the bonding curve in the creation step.
+
+### Pool
+A [pool][pool-details] is a collection of bonded currencies and their associated collateral. Each pool has a unique ID and can be managed independently. Pools can be either permissioned or trustless, as specified during the creation step.
+
+- Permissioned Pools: Only the pool manager has the authority to mint and burn bonded currencies. The manager can be updated, or the privileges can be dropped entirely.
+- Trustless Pools: Any participant can mint and burn bonded currencies without requiring special permissions.
+
+## Storage Items
+
+- `Pools`: Stores details of each pool, including its bonding curve, collateral type, and current state.
+- `NextAssetId`: Tracks the next available asset ID for new currencies.
+
+## Events
+
+- `LockSet`: Emitted when a lock is set on a pool.
+- `Unlocked`: Emitted when a pool is unlocked.
+- `PoolCreated`: Emitted when a new pool is created.
+- `RefundingStarted`: Emitted when a pool starts the refunding process.
+- `DestructionStarted`: Emitted when a pool starts the destruction process.
+- `RefundComplete`: Emitted when the refund process is complete.
+- `Destroyed`: Emitted when a pool is fully destroyed.
+- `ManagerUpdated`: Emitted when the manager of a pool is updated.
+
+## Errors
+
+- `PoolUnknown`: The specified pool ID is not registered.
+- `IndexOutOfBounds`: The specified currency index is out of bounds.
+- `NothingToRefund`: There is no collateral to refund or no remaining tokens to exchange.
+- `NoPermission`: The user does not have permission to perform the operation.
+- `PoolNotLive`: The pool is not available for use.
+- `LivePool`: The pool cannot be destroyed because there are active accounts associated with it.
+- `NotRefunding`: The operation can only be performed when the pool is in the refunding state.
+- `CurrencyCount`: The number of currencies linked to a pool exceeds the limit.
+- `InvalidInput`: The input provided is invalid.
+- `Internal`: An internal error occurred. This error should never happen.
+- `Slippage`: The cost exceeds the maximum allowed slippage.
+
+## Config Trait
+
+The `Config` trait defines the configuration parameters and associated types required by the pallet. It includes the following associated types and constants:
+
+### Associated Types
+
+- `DepositCurrency`: The currency used for storage deposits.
+- `PoolId`: The type representing the identifier for a pool.
+- `AssetId`: The type representing the identifier for an asset.
+- `CurveParameterType`: The type representing the parameters for the curve.
+- `CurveParameterInput`: The type representing the input for the curve parameters.
```

Comment:
Specify somewhere that is in unchecked maybe?

Reply:
I added more explanation in:[ea7228e](https://github.com/KILTprotocol/kilt-node/pull/807/commits/ea7228e002fa608ba2742d4b7442c92f73994433)